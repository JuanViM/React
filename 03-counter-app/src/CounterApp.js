import property from "prop-types";
import { useState } from "react";

const CounterApp= ({value = 10}) => {

const [counter,setCounter] = useState(value);

const suma = () => {
    setCounter(counter+1);
}
const resta = () => {
    setCounter(counter-1);
}
const reset = () => {
    setCounter(value);
}

    return (
        <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
        <button onClick={suma}>+1</button>
        <button onClick={resta}>-1</button>
        <button onClick={reset}>reset</button>
        </>
    )        
    
}

CounterApp.propsType={
    value: property.number
}

export default CounterApp;