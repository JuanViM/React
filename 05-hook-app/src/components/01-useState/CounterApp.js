import React, { useState } from 'react';
import "./counter.css";

export const CounterApp = () => {

    const [state, setState] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30,
        counter4: 40,
    })

    //asi desestructuramos un objeto y lo guardamos en als variables counter1 y 2
    const {counter1,counter2,counter3,counter4} = state;
  
  return (
    <>
    <h1>Counter1 = {counter1}</h1>
    <h1>Counter2 = {counter2}</h1>
    <h1>Counter3 = {counter3}</h1>
    <h1>Counter4 = {counter4}</h1>
    
    <hr />
    <button className='btn btn-primary'
    onClick={() => {
      setState({...state,//el operador spread(...) extrae todas las propiedades de state
      counter1: counter1 +1,//aqui unicamente modificamos el counter1 de todas las propiedades del spread
    }); 
    }}>+1</button>
    </>
  )
}
