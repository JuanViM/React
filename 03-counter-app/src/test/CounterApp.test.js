import CounterApp from "../CounterApp";
import React from 'react';
import {shallow} from 'enzyme';


describe('Probando la prueba de counter app', () => { 
    
    //lod efinimos aqui para que pueda ser leido dentro de todos los text
    // para asi evitar repeticion de codigo
    let wrapper = shallow(<CounterApp />);;

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    })

    test('Debe mostrar counterApp correctamente a traves de un snapshot', () => { 
           
        

        expect(wrapper).toMatchSnapshot();
     });

     test('Debe mostrar por defecto el valor de 100', () => { 
        
        const wrapper = shallow(<CounterApp value={100}/>);
        //guardamos en la variable defec lo que contenga el h2 dentro del documento
        // en este caso 100
        const defec = wrapper.find("h2").text();
        console.log(defec);
        expect(defec).toBe("100");

      });

      test('Debe de incrementar con el boton de +1', () => { 
        
        //para trabajar con los botones los encontrariamos segun su indice de aparicion
        const btn1 =  wrapper.find("button").at(0).simulate("click");
        console.log(btn1.html());
        
        const defec = wrapper.find("h2").text();

        expect(defec).toBe("11");


       });

       test('Debe resta 1', () => { 
        
        const btn2 =  wrapper.find("button").at(1).simulate("click");
        console.log(btn2.html());
        
        const defec = wrapper.find("h2").text();

        expect(defec).toBe("9");
        });

        test('El valor debe resetarse a 10', () => { 
            const wrapper = shallow(<CounterApp value={100}/>);
            const btn3 = wrapper.find("button").at(2).simulate("click");
            console.log(btn3);
            const defect = wrapper.find("h2").text();

            expect(defect).toBe("100");
         })

 })