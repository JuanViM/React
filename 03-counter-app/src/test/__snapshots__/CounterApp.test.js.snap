// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando la prueba de counter app Debe mostrar counterApp correctamente a traves de un snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</Fragment>
`;
